test/studies/adventOfCode/2022/day11/bradc/day11-par.chpl

```diff
@@ -0,0 +1,181 @@
+use IO, List, Barriers;
+
+config const numRounds = 20;
+
+// things that define a monkey
+class Monkey {
```

Comment:
I did an array of Monkey records.  Did you end up needing to have a reference/pointer to a Monkey somewhere?  Did you end up copying Monkey's around?

Reply:
I started with records, but had trouble with the I/O / inferred-array-size initialization which switching to classes resolved.  But I also like classes slightly better for this problem because (a) we've already taught them so I don't need to teach a new concept and (b) one of the rationales I give for using classes is "if the object has a unique identity and should not be copied around" which fits this situation—we don't want to clone the monkeys (though we can obviously also use records without cloning them if we're careful—I think it was the cloning that was giving me troubles with the I/O and array initialization, though... like lack of copy initializers and the like which I didn't want to have to teach).